### PR TITLE
Will you change EventInterface to public?

### PR DESCRIPTION
**Pull request by usd-yamazaki**
I want to execute this code.

    let abi = ethabi::Contract::new(ethabi::Interface::load(json.as_slice()).unwrap());
    let events = abi.events().map(|event| {
        let params = event.interface.inputs.iter().map(|event_param| {
            event_param.kind.clone()
        } ).collect::<Vec<_>>();
        (H256::from(event_signature(&event.interface.name, params.as_slice()).as_ref()), event)
    });

Thank you.

### CODE REVIEW

**Comment by rphmeier** [comment]
Would prefer to expose the fields of `EventInterface` through public getters on `Event` which gives us a lot more room for changes without breaking backwards-compatibility.

**Comment by usd-yamazaki** [comment]
Is this correct?

**Comment by rphmeier** [comment]
@usd-yamazaki Not exactly -- what I mean is that we should leave the `Event` type definition general enough to be implemented without `EventInterface` existing (since it could be removed in the future) while maintaining backwards compatibility.

For example, if you want to call `event.interface.inputs.iter()`, you could write a `fn inputs(&self) -> &[EventParam]` which does `&self.interface.inputs` internally.

Then your code becomes `event.inputs().iter()` which will behave identically.
There is already a getter for `name`.

**Comment by usd-yamazaki** [comment]
Thanks for the explanation. I think I understood it.

File: src/event.rs [hunk 1]
```diff
@@ -133,6 +134,11 @@ impl Event {
 	pub fn name(&self) -> &str {
 		&self.interface.name
 	}
+
+	/// Return the inputs of the event.
+	pub fn inputs(&self) -> &Vec<EventParam> {
```
Comment (rphmeier) [inline src/event.rs]: You can change `&Vec<EventParam>` to `&[EventParam]`. It's more idiomatic

**Comment by rphmeier** [comment]
Awesome, just one tiny nitpick and then I'm happy to merge.

**Comment by usd-yamazaki** [comment]
Thank you for telling me.
I changed it to shortened form.

**Pull request merged**